src/HDF5Msg.chpl

```diff
@@ -1782,47 +1774,31 @@ module HDF5Msg {
      */
     proc readAllHdfMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
         var repMsg: string;
-        // May need a more robust delimiter then " | "
-        var (strictFlag, ndsetsStr, nfilesStr, allowErrorsFlag, calcStringOffsetsFlag, arraysStr) = payload.splitMsgToTuple(6);
-        var strictTypes: bool = true;
-        if (strictFlag.toLower().strip() == "false") {
-          strictTypes = false;
-        }
+        var msgArgs = parseMessageArgs(payload, 7);
+        var strictTypes: bool = msgArgs.get("strict_types").getBoolValue();
 
-        var allowErrors: bool = "true" == allowErrorsFlag.toLower(); // default is false
+        var allowErrors: bool = msgArgs.get("allow_errors").getBoolValue(); // default if false
         if allowErrors {
             h5Logger.warn(getModuleName(), getRoutineName(), getLineNumber(), "Allowing file read errors");
         }
 
-        var calcStringOffsets: bool = (calcStringOffsetsFlag.toLower() == "true"); // default is false
+        var calcStringOffsets: bool = msgArgs.get("calc_string_offsets").getBoolValue(); // default is false
         if calcStringOffsets {
             h5Logger.warn(getModuleName(), getRoutineName(), getLineNumber(),
                 "Calculating string array offsets instead of reading from HDF5");
         }
 
-        // Test arg casting so we can send error message instead of failing
-        if (!checkCast(ndsetsStr, int)) {
-            var errMsg = "Number of datasets:`%s` could not be cast to an integer".format(ndsetsStr);
-            h5Logger.error(getModuleName(), getRoutineName(), getLineNumber(), errMsg);
-            return new MsgTuple(errMsg, MsgType.ERROR);
-        }
-        if (!checkCast(nfilesStr, int)) {
-            var errMsg = "Number of files:`%s` could not be cast to an integer".format(nfilesStr);
-            h5Logger.error(getModuleName(), getRoutineName(), getLineNumber(), errMsg);
-            return new MsgTuple(errMsg, MsgType.ERROR);
-        }
-
-        var (jsondsets, jsonfiles) = arraysStr.splitMsgToTuple(" | ",2);
-        var ndsets = ndsetsStr:int; // Error checked above
-        var nfiles = nfilesStr:int; // Error checked above
+        var ndsets = msgArgs.get("dset_size").getIntValue(); 
+        var nfiles = msgArgs.get("filename_size").getIntValue();
```

Comment:
I just wanna say this is great! I love how many lines or boilerplate code that we save this way